pkg/auth/providers/azure/azure_provider.go

```diff
@@ -439,3 +440,17 @@ func (ap *azureProvider) GetUserExtraAttributes(userPrincipal v3.Principal) map[
 	}
 	return extras
 }
+
+// userIDFromPrincipalID strips the leading identifier of the principal ID to extract only the ID itself.
+// For example, in a principle ID in the form of azuread_user://a1b2c3, the actual user ID is the a1b2c3 part.
+func userIDFromPrincipalID(principalID string) string {
```

Comment:
It looks like you already have a method which does a good amount of parsing for you - namely, [ParsePrincipalID](https://github.com/maxsokolovsky/rancher/blob/e848ba6dc91931e43875c88fa5ef89a55f012138/pkg/auth/providers/azure/clients/extract.go#L47). Is there a way that we can leverage this method rather than making an entirely new method, which increases our surface area for bugs in the future?